src/keysig.cpp

```diff
@@ -159,45 +204,29 @@ void KeySig::FillMap(MapOfPitchAccid &mapOfPitchAccid) const
         return;
     }
 
-    int i;
     data_ACCIDENTAL_WRITTEN accidType = this->GetAccidType();
-    for (i = 0; i < this->GetAccidCount(); ++i) {
+    for (int i = 0; i < this->GetAccidCount(); ++i) {
         mapOfPitchAccid[KeySig::GetAccidPnameAt(accidType, i)] = accidType;
     }
 }
 
-std::wstring KeySig::GetKeyAccidStrAt(int pos, data_ACCIDENTAL_WRITTEN &accid, data_PITCHNAME &pname) const
+std::optional<KeyAccidInfo> KeySig::GetKeyAccidInfoAt(int pos) const
 {
-    pname = PITCHNAME_c;
-    accid = ACCIDENTAL_WRITTEN_s;
-    std::wstring symbolStr = L"";
+    if (pos > 12) return std::nullopt;
```

Comment:
You should also check that it is not smaller than 0 - even though it should never happen.